sdk/src/Services/AutoScaling/Generated/Internal/AutoScalingAuthSchemeResolver.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Amazon.Runtime;
+using Amazon.Runtime.Credentials.Internal;
+using Amazon.Runtime.Internal;
+using Smithy.Identity.Abstractions;
+using System.Collections.Generic;
+
+// Hand-written class at the moment, generated files will include documentation for public members.
+#pragma warning disable CS1591
+
+namespace Amazon.AutoScaling.Internal
+{
+    // Auth parameters and scheme resolvers will be generated for each service;
+    // The logic for S3 and EventBridge (where authentication rules are in the endpoint-rule-set file) will be different.
+
+    public class AutoScalingAuthSchemeParameters : IAuthSchemeParameters
+    {
+        public string Operation { get; set; }
+        public string Region { get; set; }
+    }
+
+    public class AutoScalingAuthSchemeHandler : BaseAuthResolverHandler
+    {
+        /// <inheritdoc/>
+        protected override List<IAuthSchemeOption> ResolveAuthOptions(IRequestContext requestContext)
+        {
+            // Map request values to auth scheme parameters.
+            var mappedParameters = new AutoScalingAuthSchemeParameters
+            {
+                Operation = requestContext.Request.RequestName,
+                Region = requestContext.ClientConfig.RegionEndpoint?.SystemName
+            };
+
+            // TODO: In the generated version, use a singleton for the scheme resolver (similar to marshallers) to
+            // prevent creating a new instance for each request.
+            var authResolver = new AutoScalingAuthSchemeResolver();
+            return authResolver.ResolveAuthScheme(mappedParameters);
+        }
+    }
+
+    public class AutoScalingAuthSchemeResolver : IAuthSchemeResolver<AutoScalingAuthSchemeParameters>
+    {
+        public List<IAuthSchemeOption> ResolveAuthScheme(AutoScalingAuthSchemeParameters authParameters)
+        {
+            var options = new List<IAuthSchemeOption>();
+
+            switch (authParameters.Operation)
```

Comment:
Won't we have to check more then just the operation name for determining the auth? If so then a switch statement might not be a good fit. For example with EventBridge the auth changes if the EndpointId is set. Or is that type of resolution out of scope for the auth resolver.